src/Microsoft.AspNetCore.SystemWebAdapters/HttpContext.cs

```diff
@@ -41,6 +43,11 @@ public HttpContext(HttpContextCore context)
 
     public Cache Cache => _context.RequestServices.GetRequiredService<Cache>();
 
+    /// <summary>
+    /// Gets whether the current request is running in the development environment.
+    /// </summary>
+    public bool IsDebuggingEnabled => _context.RequestServices.GetRequiredService<IWebHostEnvironment>().IsDevelopment();
```

Comment:
This makes me uncomfortable but I see why it's written this way.

Reply:
Which part makes you uncomfortable?

Reply:
The way that this value was set in ASP.NET Framework is in the web.config:

```xml
  <system.web>
    <compilation debug="true" targetFramework="4.7.2">
      <assemblies>
        <add assembly="netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" />
      </assemblies>
    </compilation>
    <httpRuntime targetFramework="4.7.2" />
  </system.web>
```

there's a web.config transform that occurs on publish that removes the debug (see `Web.Release.config`):

```xml
 <system.web>
    <compilation xdt:Transform="RemoveAttributes(debug)" />
  </system.web>
```

Given that it was a static value that was mostly on dev time, having it based on the WebHostEnvironment made the most sense.

We could also consider abstracting it in some way so someone could customize it, but at this point that seems like too much effort for payoff.

Reply:
`IWebHostEnvironment.IsDevelopment()` feels closer to me because `IsDebuggingEnabled` should return true if the request is being served in debug mode, not just if there's a debugger attached to the process.

Reply:
Right, I looked and I agree. I was also going to suggest `IsDebuggingEnabled` but realized we were looking at the compilation section on .NET Framework.